From 5162a5a5d4bfddb73bbcdfec37932a8509a994b2 Mon Sep 17 00:00:00 2001
From: Pavan Nikhilesh <pbhagavatula@marvell.com>
Date: Tue, 4 Dec 2018 21:08:58 +0300
Subject: [PATCH 0849/1051] soc: octeontx2: update TIM 10ns clk source min
 interval

The hardware requires a minimum of 256 ticks (2.56us) regardless of the
number of rings in use.
Previously, it was speculated that having less number of rings (<16)
allowed a minimum of 100 ticks (1us) to be configured.

Change-Id: I106dde54b491445f8287cdf384304b364fa1a979
Signed-off-by: Pavan Nikhilesh <pbhagavatula@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/1756
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/soc/marvell/octeontx2/rvu_tim.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/soc/marvell/octeontx2/rvu_tim.c b/drivers/soc/marvell/octeontx2/rvu_tim.c
index 993aa0c44ed4..edc001a61977 100644
--- a/drivers/soc/marvell/octeontx2/rvu_tim.c
+++ b/drivers/soc/marvell/octeontx2/rvu_tim.c
@@ -190,7 +190,7 @@ int rvu_mbox_handler_tim_config_ring(struct rvu *rvu,
 
 	switch (req->clocksource) {
 	case TIM_CLK_SRCS_TENNS:
-		intervalmin = 100;
+		intervalmin = 256;
 		break;
 	case TIM_CLK_SRCS_GPIO:
 		intervalmin = 256;
-- 
2.17.1

